### PR TITLE
CTF guns only hurt mobs with CTF armor

### PR DESCRIPTION
**Pull request by coiax**
This is moving towards the possiblity of spawning CTF spawns and flag
spawns onto the station without people being caught in the crossfire.

CTF projectiles also damage CTF barriers.

### CODE REVIEW

File: code/modules/awaymissions/capture_the_flag.dm [hunk 1]
```diff
@@ -364,9 +379,25 @@
...
+	. = ..()
+
+/proc/is_ctf_target(atom/target)
+	. = FALSE
+	world << target
+	if(istype(target, /obj/structure/barricade/security/ctf))
+		. = TRUE
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		world << H.wear_suit
```
Comment (lzimann) [inline code/modules/awaymissions/capture_the_flag.dm]: debug msg

**Comment by Boredone** [comment]
If you can, I'd suggest also making it so that the guns require you wear the armor before you can use them, like with Laser Tag guns.

**Comment by Mindustry** [comment]
@Boredone don't really need that as you spawn with the armor on and cannot drop the gun anyways.

**Comment by lzimann** [comment]
it's going to be a problem if it's an admin thing, since they will spawn guns + armor and people will shoot at people wearing armor without wearing the armor(becoming immortal in CTF)

**Comment by coiax** [comment]
Unless we're getting into admin weirdness, you don't drop the CTF guns when you die as a CTF player, since they are NODROP.

**Comment by KorPhaeron** [comment]
Why are admins trying to run ctf in the middle of the station without it impacting the station

**Comment by MimicFaux** [comment]
I'd wager because of the 'cool terrain'. You see it in a lot of paintball places. Rather than an standard arena, the shooting takes place 'inside an office' or 'in a hotel'.

**Comment by Shadowlight213** [comment]
make a subtype or var. I still want to be able to hand out instagib gins at round end

**Comment by coiax** [comment]
Instagib guns aren't specific to CTF, they just gib people...

**Comment by lzimann** [comment]
honestly, I don't think we need this PR at all

**Comment by jughu** [comment]
We dont need it but it be funner having it

**Comment by optimumtact** [comment]
What, we already have laser tag, do we really need to duplicate existing functionality?

**Comment by coiax** [comment]
Laser tag is kinda lame though.

**Comment by lzimann** [comment]
can't you just use the same stuff but with actual damage instead of being stam?

**Pull request merged**